apps/src/tutorialExplorer/tutorialSet.jsx

```diff
@@ -24,8 +24,30 @@ export default class TutorialSet extends React.Component {
   tutorialClicked = item =>
     this.setState({showingDetail: true, chosenItem: item});
 
-  tutorialDetailClosed = () =>
+  tutorialDetailClosed = () => {
+    // okay, so while the tutorialDetail window is open, it's possible that the focus has changed
+    // the user could've hit left/right arrows while the detail was open and moved focus, or also could've
+    // changed focus into the dialog, so before we wipe out the chosenItem, we find the associated
+    // item in the tutorial list and focus it, and also make sure to scroll it into view because it's possible
+    // the user navigated through enough of them to be off screen.
+    const tutorialDiv = document.querySelectorAll(
+      `[data-tutorial-code="${this.state.chosenItem.code}"]`
+    )[0];
+    if (
+      document.activeElement.hasAttribute('data-tutorial-code)') &&
+      document.activeElement !== tutorialDiv
+    ) {
+      // when in doubt, set timeout! _something_ is happening here that's fiddling with the focus, so we
```

Comment:
`when in doubt, set timeout!` 
^ For my own learning, can you share some more details? This is my first time seeing setTimeout used in this way, and I'm curious to learn more (is there a link you can share for further reading)?

Reply:
Ooh! Even better. It was the `this.setState({showingDetail: false, chosenItem: null});` call that was stealing focus, and by moving that ahead of the focus/scroll combo it didn't steal it away and we could just do the focus/scroll synchronously w/o deferring to the next pass. So I dropped it.